ceph-exporter: fix systemd unit

oOHenry opened the pull request:
change systemd unit into templated unit, that the [%i](https://github.com/ceph/ceph/blob/main/systemd/ceph-exporter.service.in#L10) which is used inside the unit is working

Systemd docs: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Specifiers

For me it looks like nobody tested the exporter with the systemd unit, because it will not start with this unit

Fixes: https://tracker.ceph.com/issues/74445

<!--
  - Please give your pull request a title like

      [component]: [short description]

  - Please use this format for each git commit message:

      [component]: [short description]

      [A longer multiline description]

      Fixes: [ticket URL on tracker.ceph.com, create one if necessary]
      Signed-off-by: [Your Name] <[your email]>

    For examples, use "git log".
-->

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins test classic perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-classic/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test crimson perf` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-perf-crimson/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml)
- `jenkins test signed` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pr-commits/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-commits/config/definitions/ceph-pr-commits.yml)
- `jenkins test make check` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests/config/definitions/ceph-pull-requests.yml)
- `jenkins test make check arm64` [Jenkins Job](https://jenkins.ceph.com/job/ceph-pull-requests-arm64/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pull-requests-arm64/config/definitions/ceph-pull-requests-arm64.yml)
- `jenkins test submodules` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-pr-submodules/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-submodules/config/definitions/ceph-pr-commits.yml)
- `jenkins test dashboard` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-pull-requests/config/definitions/ceph-dashboard-pull-requests.yml)
- `jenkins test dashboard cephadm` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-dashboard-cephadm-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-dashboard-cephadm-e2e/config/definitions/ceph-dashboard-cephadm-e2e.yml)
- `jenkins test api` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-api/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-pr-api/config/definitions/ceph-pr-api.yml)
- `jenkins test docs` [ReadTheDocs](https://readthedocs.org/projects/ceph/) | [Github Workflow Definition](https://github.com/ceph/ceph/blob/main/.readthedocs.yml)
- `jenkins test ceph-volume all` [Jenkins Jobs](https://jenkins.ceph.com/view/ceph-volume%20PR/) | [Jenkins Jobs Definition](https://github.com/ceph/ceph-build/blob/main/ceph-volume-cephadm-prs/config/definitions/ceph-volume-pr.yml)
- `jenkins test windows` [Jenkins Job](https://jenkins.ceph.com/job/ceph-windows-pull-requests/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-windows-pull-requests/config/definitions/ceph-windows-pull-requests.yml)
- `jenkins test rook e2e` [Jenkins Job](https://jenkins.ceph.com/view/all/job/ceph-orchestrator-rook-e2e/) | [Jenkins Job Definition](https://github.com/ceph/ceph-build/blob/main/ceph-rook-e2e/config/definitions/ceph-orchestrator-rook-e2e.yml)
</details>

batrick commented:
Do you want this backported to `tentacle`/`squid`? If so, please create a tracker ticket:

Please create a [tracker issue](https://tracker.ceph.com/) so that this issue can be tracked for backporting.

Then, [please annotate the commit which fixes/resolves a Ceph tracker issue with](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#fixes-lines):

`Fixes: http://tracker.ceph.com/issues/...`

This is essential when examining the history of the repository (this commit fixes what) and helps merge scripts identify issues that have been resolved by a merge. See [this article](https://help.github.com/en/articles/changing-a-commit-message) on GitHub on how to amend commits and update your pull request.

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

ceph-jenkins commented:
Can one of the admins verify this patch?

tchaikov reviewed (changes requested):
we also need to update https://github.com/ceph/ceph/blob/c09cab335169c09e7a38be472097f673b645ea06/ceph.spec.in#L2153 see https://github.com/ceph/ceph/blob/c09cab335169c09e7a38be472097f673b645ea06/ceph.spec.in#L2162 for an example.

tchaikov reviewed (approved):
lgtm

tchaikov commented:
> @tchaikov thanks for bringing live in this PR agian, I adjusted ceph/ceph.spec.in, is something misssing? Do I need to create a tracker ticket?

@oOHenry hi Henry, it's up to you—see Patrick's comment at https://github.com/ceph/ceph/pull/64561#issuecomment-3410750577. However, if you don't have the bandwidth to handle this, I'm happy to take it on, as I believe this fix should be backported to the LTS branches affected by the issue your change addresses.

SrinivasaBharath commented:
@oOHenry - The builds are failing after the PR was included. Kindly look into the issue and resolve it.
Shaman build - https://shaman.ceph.com/builds/ceph/wip-bharath10-testing-2026-01-30-0333/

Commit:
ceph-exporter: fix systemd unit

change systemd unit into templated unit, that the %i which is used
inside the unit is working

Signed-off-by: Henry Hirsch <henry.hirsch@1und1.de>

oOHenry commented:
Sorry for my late reply, hope I got it right this time :)

batrick commented:
ok to test
